### PR TITLE
task/7452/create property

### PR DESCRIPTION
**Pull request by KhaiCM**
https://edu-redmine.framgia.vn/issues/7452

### CODE REVIEW

**Comment by KhaiCM** [comment]
ready

File: app/Http/Controllers/AjaxController.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+	}
+	public function getPropertyType($property_category_id, Request $request)
+	{
+		$property_type = PropertyType::where('property_category_id', $property_category_id)->get();
+		$html = '';
+		foreach($property_type as $item) {
+			$html .= '<option value="' . $item->id . '">' . $item->name . '</option>';
+		}
+		return $html;
```
Comment (BunnyPi04) [inline app/Http/Controllers/AjaxController.php]: thiếu dòng trắng trước return

File: app/Http/Controllers/PropertyController.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        }
+        $propertyCategory = PropertyCategory:: all();
+        $propertycategorys= [];
+        foreach ($propertyCategory as $item) {
+            $propertycategorys[$item->id] = $item->name;
+        }
+        $PropertyTypes = PropertyType::all();
+        $PropertyType = [];
+        foreach ($PropertyTypes as $item) {
+            $PropertyType[$item->id] = $item->name;
```
Comment (BunnyPi04) [inline app/Http/Controllers/PropertyController.php]: naming convention
xem lại cả trên nữa nhé

File: app/Http/Controllers/PropertyController.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        }
+
+        return view('fontend.properties.property_submit', compact('provinces', 'districts', 'propertycategorys','PropertyType'));
```
Comment (BunnyPi04) [inline app/Http/Controllers/PropertyController.php]: convention

File: resources/views/fontend/properties/property_submit.blade.php [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+				<li><a href="#">{{ __('label.home') }}</a></li>
+				<li class="active">{{ __('label.submit_property') }}</li>
+			</ul>
+		</div>
+	</div>
+</div>
+<!-- Sub banner 2 end -->
+<!-- User page start -->
+<div class="user-page content-area-7">
+	<div class="container">
```
Comment (BunnyPi04) [inline resources/views/fontend/properties/property_submit.blade.php]: indent = 4 spaces, xem lại cả file nhé

File: routes/web.php [hunk 5]
```diff
@@ -47,3 +47,9 @@
 Route::get('/', 'HomeController@getHome')->name('home');
 Route::get('user_page/{id}', 'UserPageController@edit')->name('user_page.edit');
 Route::put('user_page/{id}', 'UserPageController@update')->name('user_page.update');
+
+Route::get('property', 'PropertyController@create');
+Route::group(['prefix'=>'ajax'], function (){
+	Route::get('province/{province_id}', 'AjaxController@getDistrict');
```
Comment (BunnyPi04) [inline routes/web.php]: indent = 4 spaces

File: resources/views/fontend/properties/property_submit.blade.php [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+                                        {!! Form::label(__('label.describe')) !!}
+                                        {!! Form::textarea('describe', '', ['class' => 'form-control', 'id' =>'editor1', 'placeholder' => __('label.detailed_nformation')]) !!}
+                                    </div>
+                                </div>
+                            </div>
+                            <h3 class="heading">{!! Form::label('name', __('label.property_image')) !!}</h3>
+                            <div class="row mb-50">
+                                <div class="col-lg-12">
+                                    <div id="myDropZone" class="dropzone dropzone-design mb-50">
+                                        <div class="dz-default dz-message"><span>Drop files here to upload</span></div>
```
Comment (BunnyPi04) [inline resources/views/fontend/properties/property_submit.blade.php]: i18n

File: resources/views/fontend/properties/property_submit.blade.php [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+                url: "ajax/province/"+province_id,
+            }).done(function(res){
+                $("#district").html(res);
+            });
+        });
+        $("#property_category").on('change', function(){
+            var property_category_id = $(this).val();
+            $.ajax({
+                method: 'get',
+                url: "ajax/province/"+property_category_id,
```
Comment (BunnyPi04) [inline resources/views/fontend/properties/property_submit.blade.php]: convention

File: resources/views/fontend/properties/property_submit.blade.php [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+            }).done(function(res){
+                $("#property_type").html(res);
```
Comment (BunnyPi04) [inline resources/views/fontend/properties/property_submit.blade.php]: nháy đơn

**Comment by KhaiCM** [comment]
ready

File: app/Http/Controllers/PropertyController.php [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+        foreach ($province as $item) {
+            $provinces[$item->id] = $item->name;
+        }
+        $district = District::all();
+        $districts = [];
+        foreach ($district as $item) {
+            $districts[$item->id] = $item->name;
+        }
+        $propertyCategory = PropertyCategory:: all();
+        $propertycategorys= [];
```
Comment (BunnyPi04) [inline app/Http/Controllers/PropertyController.php]: xem lại cách đặt tên biến trong laravel nhé

File: public/js/ajax.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+            url: "ajax/province/"+province_id,
+        }).done(function(res){
+            $("#district").html(res);
+        });
+    });
+    $("#property_category").on('change', function(){
+        var property_category_id = $(this).val();
+        $.ajax({
+            method: 'get',
+            url: "ajax/province/"+property_category_id,
```
Comment (BunnyPi04) [inline public/js/ajax.js]: convention

File: public/js/ajax.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    $("#province").on('change', function(){
+        var province_id = $(this).val();
+        $.ajax({
+            method: 'get',
+            url: "ajax/province/"+province_id,
+        }).done(function(res){
+            $("#district").html(res);
+        });
+    });
+    $("#property_category").on('change', function(){
```
Comment (BunnyPi04) [inline public/js/ajax.js]: sử dụng nháy đơn

File: resources/lang/vi/label.php [hunk 12]
```diff
@@ -61,4 +61,23 @@
...
+    'property_type' => 'Loại tài sản',
+    'price' => 'Gía',
+    'province' => 'Tỉnh',
+    'district' => 'Huyện',
+    'describe' => 'Miêu Tả',
+    'property_image' => 'Hình Tài sản',
+    'acreage' => 'Diện tích',
+    'contact_detail' => 'Chi tiết liên hệ',
+    'home' => 'Trang chủ',
+    'submit_property' => 'Đăng tìa sản',
```
Comment (BunnyPi04) [inline resources/lang/vi/label.php]: `Đăng tài sản` nhé

File: resources/views/fontend/properties/property_submit.blade.php [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+                                        {!! Form::label(__('label.describe')) !!}
+                                        {!! Form::textarea('describe', '', ['class' => 'form-control', 'id' =>'editor1', 'placeholder' => __('label.detailed_nformation')]) !!}
+                                    </div>
+                                </div>
+                            </div>
+                            <h3 class="heading">{!! Form::label('name', __('label.property_image')) !!}</h3>
+                            <div class="row mb-50">
+                                <div class="col-lg-12">
+                                    <div id="myDropZone" class="dropzone dropzone-design mb-50">
+                                        <div class="dz-default dz-message"><span>{{__('label.drop_files_here_to_upload')}}</span></div>
```
Comment (BunnyPi04) [inline resources/views/fontend/properties/property_submit.blade.php]: convention

File: routes/web.php [hunk 14]
```diff
@@ -47,3 +47,9 @@
 Route::get('/', 'HomeController@getHome')->name('home');
 Route::get('user_page/{id}', 'UserPageController@edit')->name('user_page.edit');
 Route::put('user_page/{id}', 'UserPageController@update')->name('user_page.update');
+
+Route::get('property', 'PropertyController@create');
+Route::group(['prefix'=>'ajax'], function (){
```
Comment (BunnyPi04) [inline routes/web.php]: convention
